anchor/client/src/config.rs

```diff
@@ -131,6 +131,25 @@ pub fn from_cli(cli_args: &Anchor) -> Result<Config, String> {
      */
     config.network.listen_addresses = parse_listening_addresses(cli_args)?;
 
+    for addr in cli_args.boot_nodes_enr.clone() {
+        match addr.parse() {
+            Ok(enr) => config.network.boot_nodes_enr.push(enr),
+            Err(_) => {
+                // parsing as ENR failed, try as Multiaddr
+                // let multi: Multiaddr = addr
+                //     .parse()
+                //     .map_err(|_| format!("Not valid as ENR nor Multiaddr: {}", addr))?;
+                // if !multi.iter().any(|proto| matches!(proto, Protocol::Udp(_))) {
+                //     slog::error!(log, "Missing UDP in Multiaddr {}", multi.to_string());
+                // }
+                // if !multi.iter().any(|proto| matches!(proto, Protocol::P2p(_))) {
+                //     slog::error!(log, "Missing P2P in Multiaddr {}", multi.to_string());
+                // }
+                // multiaddrs.push(multi);
+            }
+        }
+    }
```

Comment:
This is adding the ENRs from the cli into the network config. LH does the samehttps://github.com/sigp/lighthouse/blob/stable/beacon_node/src/config.rs#L1159

Reply:
LH doing something shouldn't be used as the only source of truth to follow for `anchor`. 
In this case specifically `lighthouse` is not using `clap` derive yet (it's an ongoing process, see [1](https://github.com/sigp/lighthouse/pull/6300), [2](https://github.com/sigp/lighthouse/pull/6493)).
Anchor is using clap derive from the beginning so we don't need to iterate the cli arguments as a string separated by commas, we already have a `Vec<String>`.
What I meant in the previous comment was that we can still have `config.network.boot_nodes_enr` be `Vec<String>`, and just call
```rust
config.network.boot_nodes_enr.append(cli_args.boot_nodes_enr)
```
and then on https://github.com/sigp/anchor/pull/72/files#diff-377ff7ebcbab6b7243aedbcf1c9139b355dd5294398ca8b0b218c32ccf2f1387R121 we parse them, so that we we only iterate the enr bootnodes list once

Reply:
Thanks for clarifying, I understand now. In this case, the vector will only contain a few values. Would this optimization have a significant impact on performance?

Reply:
no worries Diego. probably it won't have a big impact on the performance but it isn't only about performance, it also makes the code more elegant at no cost

Reply:
There's also the validation perspective. It's probably a good idea to parse and check if the ENR is valid here and add only valid ones to the network config.